### PR TITLE
chore(actions): update php min supported to 8.0

### PR DESCRIPTION
**Pull request by skjnldsv**
NC 26 drops 7.4 support

- Remove 7.4 testing
- Fix and upgrade outdated psalm github action
- Update some outdated actions too

refs:
- https://github.com/nextcloud/server/pull/34997
- https://github.com/psalm/psalm-github-actions/issues/31
- https://github.com/psalm/psalm-github-actions/pull/43

### CODE REVIEW

**Comment by skjnldsv** [comment]
The `GitHub Code Scanning / Psalm` failure seems normal as I merged the github psalm workflows and it doesn't have a master base ref yet

**Comment by skjnldsv** [comment]
```
@skjnldsv skjnldsv requested review from ChristophWurst and miaulalala as code owners 12 minutes ago
@skjnldsv skjnldsv requested a review from tcitworld as a code owner 12 minutes ago
```

Wut, I did not do that !

**Comment by miaulalala** [comment]
Github rules ;)

> ```
> @skjnldsv skjnldsv requested review from ChristophWurst and miaulalala as code owners 12 minutes ago
> @skjnldsv skjnldsv requested a review from tcitworld as a code owner 12 minutes ago
> ```
> 
> Wut, I did not do that ! Why is there 45616511 reviews requests now :(

**Review by miaulalala — APPROVED** [review]
:elephant:

File: .github/workflows/lint-php.yml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+concurrency:
+  group: lint-php-${{ github.head_ref || github.run_id }}
+  cancel-in-progress: true
+
+jobs:
+  php-lint:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        php-versions: ["8.0", "8.1"]
```
Comment (nickvergessen) [inline .github/workflows/lint-php.yml:28]:
```suggestion
        php-versions: ["8.0", "8.1", "8.2"]
```

We can lint already?

Comment (skjnldsv) [reply .github/workflows/lint-php.yml:28]: Sure, I can leave the lint! :+1:

**Review by nickvergessen — APPROVED** [review]
Dislike the code changes being mixed in, but fine by me

**Comment by ChristophWurst** [comment]
> The `GitHub Code Scanning / Psalm` failure seems normal as I merged the github psalm workflows and it doesn't have a master base ref yet

Bildschirmfoto vom 2023-03-24 12-56-07

Github is not catching up and I'm unable to locate a setting to rename the reference to the old workflow file.

**Pull request merged**